{% extends 'base.html' %}
{% load widget_tweaks %}
{% load static %}

{% block content %}

<div class="container">
  <div class="docs-wrapper">
    <div id="docs-sidebar" class="docs-sidebar">
      <nav id="docs-nav" class="docs-nav navbar">
        <ul class="section-items list-unstyled nav flex-column pb-3">
          <li class="nav-item section-title">
            <a class="nav-link scrollto active" href="#section-1">
              <span class="theme-icon-holder mr-2">
                <i class="fas fa-map-signs"></i>
              </span>DeepGOWeb function predictor </a>
          </li>
          <li class="nav-item">
            <a class="nav-link scrollto" href="#item-1-1">
              Web server for submitting protein sequences</a>
          </li>
          <li class="nav-item">
            <a class="nav-link scrollto" href="#item-1-2">
              REST API </a>
          </li>
          <li class="nav-item">
            <a class="nav-link scrollto" href="#item-1-3">SPARQL query</a></li>
          <li class="nav-item section-title"><a class="nav-link scrollto active" href="#section-2"><span class="theme-icon-holder mr-2"><i class="fas fa-map-signs"></i></span>Commandline Tool</a></li>
          <li class="nav-item"><a class="nav-link scrollto" href="#item-2-1">Installation</a></li>
        </ul>
      </nav><!--//docs-nav-->
    </div><!--//docs-sidebar-->
    <div class="docs-content">
      <div class="content">
        <article class="docs-article" id="section-1">
          <header class="docs-header">
            <h1 class="docs-heading">DeepGOWeb function prediction webserver<!-- <span class="docs-time">Last updated: </span>--></h1>
            <section class="docs-intro">
              <p>DeepGOWeb is a webserver for DeepGOPlus
                protein function prediction. It allows users to obtain
                predicted protein functions in three different ways. First, 
                using the <a href="{% url "prediction"%}">Prediction</a> web page, users can
                submit protein sequences and obtain predictions which are downloadable in
                JSON format. Second, DeepGOWeb provides a <a href="{%
                url "api"%}">REST
                  API</a> for users to access our servers
                programmatically. Finally, users can use our
                <a href="{% url "sparql"%}">SPARQL
                endpoint</a> to call DeepGOPlus within a SPARQL query.</p>
            </section><!--//docs-intro-->
            <section class="docs-section" id="item-1-1">
	      <h2 class="section-heading">Web page for submitting
	        protein sequences</h2>
                  <p>
                    Users should provide the following data to use the
                    service on the <a href="{% url
                    "prediction"%}">Prediction</a> page:
                    <ul>
                      <li>Format: <strong>FASTA</strong> format or <strong>Raw
                          Sequences</strong> separated by a newline.</li>
                      <li>Threshold: a value between 0.1 and 1.0 for filtering
                        predictions by the confidence score of the model.</li>
                      <li>Data: Protein sequences in the selected
                      format. A maximum of 10 sequences are allowed in one request.</li>
                    </ul>
                  </p>
              <p>
                After submitting the request, users will be redirected
                to results page where they can see the predictions and
                download them in JSON format. Users can also save the
                link to the results page and come back to it anytime. 
              </p>
            </section><!--//section-->
	    <section class="docs-section" id="item-1-2">
	      <h2 class="section-heading">REST API</h2>
              <p>The REST API allows accessing the DeepGOWeb service
              programmatically. Here we provide an example using
              python
              and <a href="https://requests.readthedocs.io/en/master/">Requests</a>
                library.</p>
            <div class="docs-code-block">
              <pre>import requests

sequence = "MPYKLKKEKEPPKVAKCTAKPSSSGKDGGGENTEEAQPQPQPQPQPQAQSQPPSSNKRPSNSTPPPTQLSKIKYSGGPQIVKKERRQSSSRFNLSKNRELQKLPALKDSPTQEREELFIQKLRQCCVLFDFVSDPLSDLKFKEVKRAGLNEMVEYITHSRDVVTEAIYPEAVTMFSVNLFRTLPPSSNPTGAEFDPEEDEPTLEAAWPHLQLVYEFFLRFLESPDFQPNIAKKYIDQKFVLALLDLFDSEDPRERDFLKTILHRIYGKFLGLRAYIRRQINHIFYRFIYETEHHNGIAELLEILGSIINGFALPLKEEHKMFLIRVLLPLHKVKSLSVYHPQLAYCVVQFLEKESSLTEPVIVGLLKFWPKTHSPKEVMFLNELEEILDVIEPSEFSKVMEPLFRQLAKCVSSPHFQVAERALYYWNNEYIMSLISDNAARVLPIMFPALYRNSKSHWNKTIHGLIYNALKLFMEMNQKLFDDCTQQYKAEKQKGRFRMKEREEMWQKIEELARLNPQYPMFRAPPPLPPVYSMETETPTAEDIQLLKRTVETEAVQMLKDIKKEKVLLRRKSELPQDVYTIKALEAHKRAEEFLTASQEAL"
threshold = 0.3
r = requests.post('http://deepgoplus.bio2vec.net/deepgo/api/create', data={'data_format': 'enter', 'data': sequence, 'threshold': threshold})                
result = r.json()
              </pre>
            </div>
            
	    </section><!--//section-->
	    <section class="docs-section" id="item-1-3">
	      <h2 class="section-heading">SPARQL</h2>
    <p>
      The SPARQL endpoint allows to call the DeepGOPlus function
      prediction model within a SPARQL query. We provide a custom SPARQL function called "deepgo" which
      takes a protein sequence and prediction threshold as an input and
      returns the predicted functions along with the subontology, label,
      and prediction score. The output can be downloaded in different
      formats such as json, xml, csv or text.
    </p>
    <p>
      Example queries:
      <ul>
        <li>
          Example 1: Simple example query
          <pre><xmp>PREFIX dg: <http://deepgoplus.bio2vec.net/functions#>
PREFIX GO: <http://purl.obolibrary.org/obo/GO_> 

SELECT ?ont ?go ?label ?score
{
 (?ont ?go ?label ?score)
		    dg:deepgo("MPYKLKKEKEPPKVAKCTAKPSSSGKDGGGENTEEAQPQPQPQPQPQAQSQPPSSNKRPSNSTPPPTQLSKIKYSGGPQIVKKERRQSSSRFNLSKNRELQKLPALKDSPTQEREELFIQKLRQCCVLFDFVSDPLSDLKFKEVKRAGLNEMVEYITHSRDVVTEAIYPEAVTMFSVNLFRTLPPSSNPTGAEFDPEEDEPTLEAAWPHLQLVYEFFLRFLESPDFQPNIAKKYIDQKFVLALLDLFDSEDPRERDFLKTILHRIYGKFLGLRAYIRRQINHIFYRFIYETEHHNGIAELLEILGSIINGFALPLKEEHKMFLIRVLLPLHKVKSLSVYHPQLAYCVVQFLEKESSLTEPVIVGLLKFWPKTHSPKEVMFLNELEEILDVIEPSEFSKVMEPLFRQLAKCVSSPHFQVAERALYYWNNEYIMSLISDNAARVLPIMFPALYRNSKSHWNKTIHGLIYNALKLFMEMNQKLFDDCTQQYKAEKQKGRFRMKEREEMWQKIEELARLNPQYPMFRAPPPLPPVYSMETETPTAEDIQLLKRTVETEAVQMLKDIKKEKVLLRRKSELPQDVYTIKALEAHKRAEEFLTASQEAL" 0.3) .
}
    </xmp></pre>
        </li>
        <li>Example 2: Federated query which runs
        DeepGOPlus on two sequences
        from the <a href="https://sparql.uniprot.org"> UniProt SPARQL
            Endpoint</a>
         <pre><xmp>PREFIX dg: <http://deepgoplus.bio2vec.net/functions#>
PREFIX GO: <http://purl.obolibrary.org/obo/GO_> 
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>
SELECT ?protein ?organism ?isoform ?sub ?go ?label ?score
WHERE
{
{
SELECT DISTINCT ?protein ?organism ?isoform ?aa_sequence
  WHERE 
  {
  SERVICE <http://sparql.uniprot.org/sparql> {
    ?protein a up:Protein .
    ?protein up:organism ?organism .
    ?organism rdfs:subClassOf taxon:9606 .
    ?protein up:sequence ?isoform .
    ?isoform rdf:value ?aa_sequence .
  }
  }
LIMIT 2
}
(?sub ?go ?label ?score) dg:deepgo(?aa_sequence 0.3) .
}
    </xmp></pre>
        </li>
      </ul>
    </p>
	    </section><!--//section-->
        </article>
        <article class="docs-article" id="section-2">
          <header class="docs-header">
            <h1 class="docs-heading">Commandline tool</h1>
            <section class="docs-intro">
              <p>Users can also install DeepGOPlus on their system
              and run DeepGOPlus locally. This is useful when
              prediction functions for a large amount of protein sequences.</p>
            </section><!--//docs-intro-->
          </header>
          <section class="docs-section" id="item-2-1">
            <h2 class="section-heading">Installation</h2>
            <h3>pip</h3>
            <div class="docs-code-block">
              <code>pip install deepgoplus</code>
            </div>
            <h5 class="pt-4">Download the data</h5>
            <p>Download all the files from <a href="http://deepgoplus.bio2vec.net/data/data.tar.gz">http://deepgoplus.bio2vec.net/data/data.tar.gz</a> and extract them into data folder</p>
            <p>Run:</p>
            <div class="docs-code-block">
              <code class="shell">deepgoplus
              --data-root path_to_data_folder --in-file
              input_fasta_filename </code>
            </div>
            </section><!--//section-->
        </article><!--//docs-article-->
      </div> 
    </div>
  </div><!--//docs-wrapper-->
</div>

<script src="{% static "yarn/jquery/dist/jquery.min.js" %}"></script>
<script src="{% static "yarn/popper.js/dist/umd/popper.min.js" %}"></script>
<script src="{% static "yarn/bootstrap/dist/js/bootstrap.min.js" %}"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.8/highlight.min.js"></script>
<script src="{% static "yarn/jquery.scrollto/jquery.scrollTo.min.js" %}"></script>
<script src="{% static "yarn/ekko-lightbox/dist/ekko-lightbox.min.js" %}"></script> 
<script src="{% static "js/doc.js" %}"></script> 
{% endblock %}
